### PR TITLE
Fix a TF Vision Encoder Decoder test

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Simple fix for the following CI failure: 

```
tests.vision_encoder_decoder.test_modeling_tf_vision_encoder_decoder.TFViT2GPT2EncoderDecoderModelTest.test_pt_tf_equivalence
```
This is due to a few PyTorch device issue (need `pt_inputs` to be on the correct device).

Tested on GCP GPU VM - it pass now.

@patrickvonplaten @sgugger

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thnaks for fixing!

**Comment by ydshieh** [comment]
Probably wait @patrickvonplaten too, but once ready, should I merge this PR myself ? (or still should be one of you to do so?)

**Comment by sgugger** [comment]
You can merge your own PRs as long as you have the approvals you want :-)

**Review by patrickvonplaten — APPROVED** [review]

File: tests/vision_encoder_decoder/test_modeling_tf_vision_encoder_decoder.py [hunk 1]
```diff
@@ -321,7 +324,7 @@ def check_pt_tf_equivalence(self, pt_model, tf_model, inputs_dict):
         self.assertEqual(len(tf_outputs), len(pt_outputs), "Output lengths differ between TF and PyTorch")
 
         for tf_output, pt_output in zip(tf_outputs, pt_outputs):
-            self.assert_almost_equals(tf_output.numpy(), pt_output.numpy(), 1e-3)
+            self.assert_almost_equals(tf_output.numpy(), pt_output.detach().to("cpu").numpy(), 1e-3)
```
Comment (patrickvonplaten) [inline tests/vision_encoder_decoder/test_modeling_tf_vision_encoder_decoder.py:327]: note that `detach()` is not needed here since it's wrapped in a `torch.no_grad()` but doesn't really matter so ok for me
